libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m

```diff
@@ -159,9 +159,11 @@ - (NSURLRequest *)prepareRequestForSend {
         // Sets HTTP method on the request.
         [self.request setHTTPMethod:[SFRestRequest httpMethodFromSFRestMethod:self.method]];
 
-        // Sets OAuth Bearer token header on the request.
-        NSString *bearer = [NSString stringWithFormat:@"Bearer %@", user.credentials.accessToken];
-        [self.request setValue:bearer forHTTPHeaderField:@"Authorization"];
+        // Sets OAuth Bearer token header on the request (if not already present).
+        if (self.request.allHTTPHeaderFields && self.request.allHTTPHeaderFields.allKeys && ![self.request.allHTTPHeaderFields.allKeys containsObject:@"Authorization"]) {
```

Comment:
Minor enhancement: The `Bearer` token header should be set only if it's not already set by the client. Current user's token should be the fallback if not set - it shouldn't overwrite a token that's being sent in by the client.